### DISCUSSION TITLE
Trouble Loading TGA Image from Android path in C++ with Magick++: "No image was loaded" Exception

### DISCUSSION THREAD

**Discussion by ramasita** [ImageMagick/ImageMagick#6994] [General]
Hello everyone,

I'm facing an issue with the Magick++ library in C++. I'm trying to load a TGA image from android device that I've successfully written to disk using Magick++ but when I attempt to read the same image, I encounter an exception stating "No image was loaded". 
Here's the snippet of my code where the issue occurs:

```
Magick::Image image1, image2;
image1.read(imagePath1); // exception thrown here
image2.read(imagePath2);
```

The variable imagePath1 points to the path where the image is saved (_/data/local/temp/abc/image1.tga_). The file image1.tga seems to be correctly formatted, as per the following checks:

> File type (using the file command): "Targa image data - RGB 800 x 800 x 24 - top"
> Image details (using the identify command): "TGA 800x800 800x800+0+0 8-bit sRGB 1.8311MiB 0.000u 0:00.002"

Despite the image being seemingly correctly saved and verified, when I try to read it back using Magick++, it throws an exception. Could anyone help me understand why this issue is occurring with an image written and read using Magick++ from the same location and path?

Any insights or suggestions would be greatly appreciated!

**Comment by fmw42** [comment]
What was the exact error message.  What is your IM version and OS/Platform (assumed Windows from file path)

**Comment by ramasita** [comment]
C++ exception - "No image was loaded."
IM version: ImageMagick-7.1.1-24
OS: Android
Platform: arm64

Image was written by Magick++ from raw memory. I cannot read back the same image from the same path in the android device.

**Reply by fmw42** [reply]
Are you sure your path is correct and file name is correct?

**Reply by ACmodel** [reply]
May I ask if it was finally resolved? I have also encountered this problem on my end and I would like to ask how you solved it. Could you please reply to me.

**Comment by ramasita** [comment]
Yes, it’s an android disk path where I saved the image to.

**Reply by fmw42** [reply]
But that may not be correct for Imagemagick.  Try the relative path to the file from your current working directory

**Reply by ramasita** [reply]
My logic actually requires dumping the image to disk from raw memory and then use it to compare it against the image(2) pushed to the device alongside.

**Comment by snibgo** [comment]
I would suggest you post the tga file here so we can see if it is weird. But you say `identify` can read it without problem, so I expect the file is fine.

Perhaps there is a bug in your program, like it doesn't close the file before reading it.
